src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/Zaimanhua.kt

```diff
@@ -203,6 +219,63 @@ class Zaimanhua : HttpSource(), ConfigurableSource {
         }
     }
 
+    private fun commentsInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+        if (request.tag(String::class) != COMMENTS_FLAG) {
+            return response
+        }
+        val comments = parseChapterComments(response, COMMENT_MAX_HEIGHT / (COMMENT_UNIT * 2) - 1).toMutableList()
+        comments.add(0, "章末吐槽：")
+
+        val paint = TextPaint().apply {
+            color = Color.BLACK
+            textSize = COMMENT_UNIT_F
+            isAntiAlias = true
+        }
+
+        var height = COMMENT_UNIT
+        val layouts = comments.map {
+            @Suppress("DEPRECATION")
+            StaticLayout(it, paint, COMMENT_WIDTH - 2 * COMMENT_UNIT, Layout.Alignment.ALIGN_NORMAL, 1f, 0f, false)
+        }.takeWhile {
+            val lineHeight = it.height + COMMENT_UNIT
+            if (height + lineHeight <= COMMENT_MAX_HEIGHT) {
+                height += lineHeight
+                true
+            } else {
+                false
+            }
+        }
+
+        val bitmap = Bitmap.createBitmap(COMMENT_WIDTH, height, Bitmap.Config.ARGB_8888)
+        bitmap.eraseColor(Color.WHITE)
+        val canvas = Canvas(bitmap)
+
+        var y = COMMENT_UNIT
+        for (layout in layouts) {
+            canvas.save()
+            canvas.translate(COMMENT_UNIT_F, y.toFloat())
+            layout.draw(canvas)
+            canvas.restore()
+            y += layout.height + COMMENT_UNIT
+        }
+
+        val output = ByteArrayOutputStream()
+        bitmap.compress(Bitmap.CompressFormat.PNG, 0, output)
+        val body = output.toByteArray().toResponseBody("image/png".toMediaType())
+        return response.newBuilder().body(body).build()
+    }
+
+    override fun imageRequest(page: Page): Request {
+        if (page.url == COMMENTS_FLAG) {
+            return GET(page.imageUrl!!, apiHeaders).newBuilder()
+                .tag(String::class, COMMENTS_FLAG)
```

Comment:
You can use `url.pathSegments[n] == "viewpoint"` to pinpoint the comment requests.

Reply:
> You can use `url.pathSegments[n] == "viewpoint"` to pinpoint the comment requests.

Using `COMMENTS_FLAG` is simpler, as it requires no additional modifications here even if the interface changes. Furthermore, `page.url` here is a string, not a URL object. To use `pathSegments[n] == “viewpoint”`, we must convert it to URL and check that `pathSegments` is not less than `n` before `pathSegments[n] == “viewpoint”`.

Reply:
> Furthermore, `page.url` here is a string, not a URL object.

I mean checking `request.url.pathSegments` in the interceptor.

> Using `COMMENTS_FLAG` is simpler, as it requires no additional modifications here even if the interface changes.

Not quite sure about that, but I'll merge it regardless.

You're using 2 different ways to tag pages: `Page.url == COMMENTS_FLAG` to tag comments and `Page.imageUrl.fragment != null` to tag retries, which looks very dirty.